src/components/FirstRegistrationForm.tsx

```diff
@@ -0,0 +1,95 @@
+import { Formik, Form, Field } from 'formik';
+import Email from "./Email";
+import PasswordAndConfirm from './PasswordAndConfirm';
+import Checkbox from './Checkbox';
+import './FirstRegistrationForm.scss';
+import { isEmailExists } from '../services/api/isEmailExists';
+
+interface FormValues {
+  email: string;
+  password: string;
+  confirmPassword: string;
+  termsAndConditions: boolean;
+}
+
+interface FormErrors {
+  [key: string]: any
+}
+
+const emailInvalidationRules = [
+  /^\s*$/, // check string not empty
+  /^[^@]+$/, // @ should exist
+  /@[^@]*@/, // onle one @ is admissible
+  /^\./, // '.' can't be first symbol
+  /\.{2,}.+(?=@)/, // '.' can't repeat more than once in a row
+  /\.(?=@)/, // '.' can't be before @
+  /[^A-Za-z0-9_!#$%&'.*+\-/=?^`{|}~].*(?=@)/, // include only valid symbols before @
+  /(?<=@).*[^a-z0-9\-.]/, // include only valid symbols before @
+];
+
+const minSymbol = 6;
+const maxSymbol = 256;
+const passwordRegex = new RegExp("^[-/=!#$%&'*+?^_`{|}~.A-Z0-9]{" + minSymbol + "," + maxSymbol + "}$", "i");
+
+const FirstRegistrationForm = () => {
+  const initialValues: FormValues = {
+    email: '',
+    password: '',
+    confirmPassword: '',
+    termsAndConditions: false
+  }
+
+  const validate = async (values: FormValues) => {
+    const isEmailExistsInDB = await isEmailExists(values.email);
+
+    let errors: FormErrors = {};
+    if (isEmailExistsInDB) {
+      errors.email = 'This email address is already registered';
+    }
+    if (emailInvalidationRules.some(rule => rule.test(values.email))) {
+      errors.email = 'Please enter a valid email address';
+    }
+    if (!passwordRegex.test(values.password)) {
+      errors.password = `An invalid character is present in the password. Password must be between ${minSymbol} and ${maxSymbol} characters; upper or lower case Latin letters (a–z, A–Z);numbers from 0 to 9;symbols ! # $ % & ' * + - / = ? ^ _ \` { | } ~`;
```

Comment:
Password **length** must be

Please add spaces after ; in your text:

 upper or lower case Latin letters (a–z, A–Z)**;numbe**rs from 0 to **9;sy**mbols

Reply:
Error text taken from requirements:
<img width="369" alt="image" src="https://user-images.githubusercontent.com/47752551/197391394-96aa8268-de4f-4d50-8193-90ed5898aadf.png">

Reply:
You can resolve this conversation